src/main/java/com/hazelcast/gcp/GcpClient.java

```diff
@@ -79,16 +114,29 @@ public String call() {
         }, RETRIES));
     }
 
-    List<GcpAddress> getAddresses() {
+    List<GcpAddress> getAddresses() throws Exception {
+        // First check to see if service account has proper permissions to fetch addresses - if not, don't retry call
+        try {
+            return fetchGcpAddresses();
```

Comment:
The same, I'd avoid making a GCP Call just to check if we have access rights. It makes always a redundant HTTP call.